libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/tripsession/LocationPermissionComponent.kt

```diff
@@ -0,0 +1,107 @@
+package com.mapbox.navigation.dropin.component.tripsession
+
+import android.Manifest
+import android.content.Context
+import androidx.activity.ComponentActivity
+import androidx.activity.result.ActivityResultCallback
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.repeatOnLifecycle
+import com.mapbox.android.core.permissions.PermissionsManager
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.lifecycle.UIComponent
+import com.mapbox.navigation.utils.internal.logW
+import kotlinx.coroutines.launch
+import java.lang.ref.WeakReference
+
+/**
+ * Default experience for location permissions.
+ *
+ * @param componentActivityRef used for requesting location permissions
+ * @param tripSessionStarterViewModel used to notify when location permissions are granted
+ */
+internal class LocationPermissionComponent(
+    private val componentActivityRef: WeakReference<ComponentActivity>?,
+    private val tripSessionStarterViewModel: TripSessionStarterViewModel,
```

Comment:
It feels a bit weird to have `tripSessionStartedViewModel` injected into `PermissionComponent`, even though I understand why is it being done. Just 🤔  if we should have some sort of interface
```
interface OnPermissionsUpdated {
    fun onPermissionsGranted(isGranted: Boolean)
}
```
that `TripSessionStarterViewModel` could hook onto

Reply:
Resolving this based on internal discussion for this comment.